tests/python/test_get_external_tensor_shape.py

```diff
@@ -0,0 +1,33 @@
+import taichi as ti
+import numpy as np
+
+
+@ti.all_archs
+def test_get_external_tensor_shape_access():
+    @ti.kernel
+    def func(x: ti.ext_arr(), index: ti.template()) -> ti.i32:
+        return x.shape[index]
+
+    size = np.random.randint(low=1, high=10, size=(8, ))
```

Comment:
Let's use a deterministic `size` here (e.g. 7). If a unit test fails, we want it to fail deterministically :-)

Reply:
Yes, please consider make use of `@pytest.mark.parametrize` and use a list of representitive shapes as parameter, e.g.:
- 3
- (3, )
- (5, 6)
- (3, 1, 5)

Reply:
@yuanming-hu, @archibate I've now made the arrays fixed size and initialised with fixed values relative to the minimum number of bits an `int` can have as specified [here](https://en.cppreference.com/w/cpp/language/types). 

Note I've squashed my commits in this branch however the original version can still be seen [here](https://github.com/thinking-tower/taichi/pull/2/commits).

Reply:
In hindsight, I realized that it's possible to run out of CUDA memory if the arrays are too big. I've reduced the size of the tests and observed that the test assume that you have a NVIDIA GPU with > `2.5`GB. In #356, it's mentioned that taichi won't support "pre-Pascal GPUs". In the [wikipedia page](https://en.wikipedia.org/wiki/GeForce_900_series#GeForce_900_(9xx)_series), there is a GPU with memory `1024 MB`. Would this be an issue?

Reply:
We actually do support pre-Pascal GPUs as long as you set `TI_USE_UNIFIED_MEMORY=0`. For test cases, we simply always limited memory usage < 256MB :-) Your current implementation looks good.